libGitWrap/Private/DiffPrivate.hpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * libGitWrap - A Qt wrapper library for libgit2
+ * Copyright (C) 2015 The MacGitver-Developers <dev@macgitver.org>
+ *
+ * (C) Nils Fenner <nils@macgitver.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License (Version 2) as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include "libGitWrap/Private/IndexPrivate.hpp"
+#include "libGitWrap/Private/RepoObjectPrivate.hpp"
+#include "libGitWrap/Private/TreePrivate.hpp"
+
+
+namespace Git
+{
+
+    class Diff;
+
+    namespace Internal
+    {
+
+        class DiffOptions : public QSharedData
```

Comment:
In this class, can you please avoid the 3-fold indirection? Since it's an internal class, we can deal with size changes through recompile. There's no need to have the `git_diff_options` in a separate heap block.

That would also avoid the `init` parameter on the second constructor. If there's options going into it, we don't need to initialise...